microprofile/metrics-smallrye/src/main/java/org/wildfly/extension/microprofile/metrics/MicroProfileMetricsExtension.java

```diff
@@ -74,18 +76,23 @@ static ResourceDescriptionResolver getResourceDescriptionResolver(final boolean
         return new StandardResourceDescriptionResolver(prefix.toString(), RESOURCE_NAME, MicroProfileMetricsExtension.class.getClassLoader(), true, useUnprefixedChildTypes);
     }
 
+    public MicroProfileMetricsExtension() {
+        super(EXTENSION_NAME, SUBSYSTEM_NAME);
+    }
+
     @Override
-    public void initialize(ExtensionContext context) {
+    protected Set<ManagementResourceRegistration> initializeLegacyModel(final ExtensionContext context) {
         final SubsystemRegistration subsystem = context.registerSubsystem(SUBSYSTEM_NAME, CURRENT_MODEL_VERSION);
         subsystem.registerXMLElementWriter(CURRENT_PARSER);
 
         final ManagementResourceRegistration registration = subsystem.registerSubsystemModel(new MicroProfileMetricsSubsystemDefinition());
-        registration.registerOperationHandler(GenericSubsystemDescribeHandler.DEFINITION, GenericSubsystemDescribeHandler.INSTANCE);
+
+        return Collections.singleton(registration);
     }
 
     @Override
-    public void initializeParsers(ExtensionParsingContext context) {
-        context.setSubsystemXmlMapping(SUBSYSTEM_NAME, MicroProfileMetricsParser_1_0.NAMESPACE, MicroProfileMetricsParser_1_0::new);
+    public void initializeLegacyParsers(ExtensionParsingContext context) {
+        context.setSubsystemXmlMapping(SUBSYSTEM_NAME, MicroProfileMetricsParser_1_0.NAMESPACE, new MicroProfileMetricsParser_1_0());
```

Comment:
This should stay the way it was. A supplier ends up being created eventually (from a lambda) so moving to the constructor call saves nothing. And the standard idiom is to use a supplier for all but the current parser (and perhaps for it as well.) In some cases that results in a minor perf improvement, hence the standard idiom.

Reply:
Reverted